docs/sources.md

```diff
@@ -0,0 +1,246 @@
+You can override this behavior by setting `keepGitDir: true` in the git configuration.
+
+Git repositories are considered to be "directory" sources.
+
+
+### HTTP
+
+HTTP sources fetch a file from an HTTP URL.
+HTTP content is not verified by digest today, but it is in the roadmap.
+
+```yaml
+sources:
+  someSource1:
+    http:
+      # No Digest verification
+      url: https://example.com/someFile.txt
+```
+
+The HTTP source type is considered to be a "file" source.
+
+### Build context
+
+Clients provide a build context to Dalec.
+As an  example, here is how the Docker client provides a build context to Dalec:
+
+```bash
+$ docker build <some args> .
+```
+
+In this case the `.`, or current directory, is the build context.
+Dalec is able to use the build context as a source:
+
+```yaml
+sources:
+  someSource:
+    context: {}
+```
+
+Note the empty brackets.
+This is an unfortunate syntax requirement to not have `context` considered as a nil value.
+This is the equivelent of the following:
+
+```yaml
+sources:
+  someSource:
+    context:
+      name: "context"
+```
+
+Where `name: "context"`, not to be confused with the source type `context`, is named by convention by the docker CLI.
+Additionally contexts can be passed in from the docker cli: `docker build --build-context <name>=<path>`.
+The `<name>` would be the name to use in your yaml to access it.
+
+The `context` source type is considered to be a "directory" source.
+
+### Inline
+
+Inline sources are sources that are defined inline in the Dalec configuration.
+You can only specify one of `file` or `dir` in an inline source.
+Directories cannot be nested in inline sources.
+Filenames must not contain a path separator (`/`).
+
+```yaml
+sources:
+  someInlineFile:
+    inline:
+      # This is the content of the source.
+      file:
+        uid: 0
+        gid: 0
+        permissions: 0644
+        contents: |
+          some content
+          some more content
+
+  someInlineDir:
+    inline:
+      dir:
+        uid: 0
+        gid: 0
+        permissions: 0755
+        files:
+          # This is the content of the source.
+          file1:
+            contents: |
+              some content
+              some more content
+            permissions: 0644
+          file2:
+            contents: |
+              some content
+              some more content
+            permissions: 0644
+```
+
+Inline sources with `file` are considered to be "file" sources.
+Inline sources with `dir` are considered to be "directory" sources.
+
+### Docker Image
+
+Docker image sources fetch a docker image from a registry.
+The output of this source is a directory containing the contents of the image.
+
+```yaml
+sources:
+  someDockerImage:
+    image:
+      ref: docker.io/library/alpine:3.14
+```
+
+You can also run commands in the image before fetching the contents.
+This is especially useful for generating content.
```

Comment:
Not sure I understand this. are you running the commands inside the image you are fetching? and what content are you retrieving?

Reply:
maybe, show a possible real use scenario?

Reply:
Good idea, the examples I gave are not actually useful in the context of the spec.

Reply:
Done.